features/lorawan/lorastack/phy/LoRaPHYAU915.cpp

```diff
@@ -486,13 +486,13 @@ uint8_t LoRaPHYAU915::link_ADR_request(adr_req_params_t* params,
     // Update channelsMask if everything is correct
     if (status == 0x07) {
         // Copy Mask
-        copy_channel_mask(channel_masks, temp_channel_masks, AU915_CHANNELS_MASK_SIZE);
+        copy_channel_mask(channel_mask, temp_channel_masks, AU915_CHANNEL_MASK_SIZE);
 
-        current_channel_masks[0] &= channel_masks[0];
-        current_channel_masks[1] &= channel_masks[1];
-        current_channel_masks[2] &= channel_masks[2];
-        current_channel_masks[3] &= channel_masks[3];
-        current_channel_masks[4] = channel_masks[4];
+        current_channel_mask[0] &= channel_mask[0];
+        current_channel_mask[1] &= channel_mask[1];
+        current_channel_mask[2] &= channel_mask[2];
+        current_channel_mask[3] &= channel_mask[3];
+        current_channel_mask[4] = channel_mask[4];
```

Comment:
Not sure what this is doing, but whatever it is, word 4 is inconsistent.

Reply:
First 64 channels are upstream only. By ANDing the mask provided by the network and the mask which was used for last transmission we make sure that the we do not use the channels marked 'used' in current_channel_mask even if the network asks us to enable all channels. Last 8 channels are downstream only and hence we update the last block as it is.

Reply:
^ Would be a good comment to add in code.

Reply:
I will address that if there will be a next iteration. Otherwise, in the next release. Noted.